lib/AsyncUtils.lua

```diff
@@ -29,6 +29,37 @@ function AsyncUtils.parallel(things)
     return Promise.all(promises)
 end
 
+--[[
+    Given a table, this function returns a promise which
+    resolves once all of the table values have resolved, or rejects
+    if any of the array elements reject.
+
+    Any values in the table which aren't promises are considered
+    resolved immediately.
+
+    The promise resolves to a table mapping the input table keys to resolved values.
+]]
+function AsyncUtils.props(things)
+    local keys = TableUtils.Keys(things)
```

Comment:
looks good, just needs some validation and tests (which you mentioned are on your list) 👍

Reply:
needs assertions